### PR TITLE
Fix auto-select on model failure

### PR DESCRIPTION
**Pull request by prasankh**
[ODSC-63260](https://jira.oci.oraclecorp.com/browse/ODSC-63260)
Improved auto-select robustness by selecting the best among successful models.

### CODE REVIEW

File: ads/opctl/operator/lowcode/forecast/model_evaluator.py [hunk 1]
```diff
@@ -147,10 +150,12 @@ def find_best_model(self, datasets: ForecastDatasets, operator_config: ForecastO
             model = SupportedModels.Prophet
             logger.error(f"Running {model} model as auto-select failed with the following error: {e.message}")
             return model
-        avg_backtests_metrics = {key: sum(value.values()) / len(value.values()) for key, value in metrics.items()}
-        best_model = min(avg_backtests_metrics, key=avg_backtests_metrics.get)
+        nonempty_metrics = {model: metric for model, metric in metrics.items() if metric != {}}
+        avg_backtests_metric = {model: sum(value.values()) / len(value.values())
+                                for model, value in nonempty_metrics.items()}
+        best_model = min(avg_backtests_metric, key=avg_backtests_metric.get)
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/model_evaluator.py:156]: What if the metric is set to accuracy "r-squared"? Do we prevent this?

**Review by ahosler — APPROVED** [review]
looks good! Couple quick questions

File: ads/opctl/operator/lowcode/forecast/utils.py [hunk 2]
```diff
@@ -19,10 +19,9 @@
     mean_absolute_percentage_error,
     mean_squared_error,
 )
-try:
-    from scipy.stats import linregress
-except:
-    from sklearn.metrics import r2_score
+
+from scipy.stats import linregress
```
[the remark this answers is not in the document]
Comment (prasankh) [reply ads/opctl/operator/lowcode/forecast/utils.py:23]: Not explicitly, statsmodels is there and it is dependent on scipy.

File: ads/opctl/operator/lowcode/forecast/model_evaluator.py (hunk 1 above)
Comment (prasankh) [reply ads/opctl/operator/lowcode/forecast/model_evaluator.py:156]: The r-squared metric is not present in forecast/schema.yaml file as an allowed optimization metric, so it is not used for selecting the best model. Although it gets calculated when test data is provided only for reporting purpose.

**Review by codeloop — APPROVED** [review]
